src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -220,7 +244,10 @@ bool convert_function_precision(const std::shared_ptr<Model>& f,
     // otherwise we insert Convert operation.
     auto ops = f->get_ordered_ops();
     for (auto& node : ops) {
-        if (skip_precision_sensitive && fp16_compression_is_disabled(node) && has_fp16_compression)
+        // This is a hotfix that checks if FROM types are not different from the
+        // node's output element type. To be investigated more in CVS-155990
+        if ((skip_precision_sensitive && fp16_compression_is_disabled(node) && has_fp16_compression) ||
+            !node_output_type_match(node, precisions))
```

Comment:
Checking only output types is not enough. Inputs also may have unsupported precisions. It will not work particularly for Select when we are doing boolean -> u8.